### PR TITLE
Add Azure as build provider

### PR DESCRIPTION
**Pull request by gjolly**
Because why not?

It can be actually useful to avoid having to upload Azure images (30GB, sparse) from my personal laptop to my Azure storage account. If using `--no-cleanup`, then connecting to the VM when the build is done, one can directly upload the images from Azure to Azure.

### CODE REVIEW

**Comment by toabctl** [comment]
@gjolly can you add some documentation to the readme with an example how to use it?

**Comment by gjolly** [comment]
I will add a dedicated example for Azure but it general it works the same way as AWS or multipass with `--build-provider azure`

**Comment by toabctl** [comment]
I would like to see proper commit messages including **why** you are doing things.

**Comment by gjolly** [comment]
@toabctl I kept the commits but documented them.

**Comment by CodeMouse92** [comment]
Can this be wrapped up and landed? Because I can't access AWS, I need this for Bartender post-haste.

File: scripts/ubuntu-bartender/azure-provider [hunk 1]
```diff
@@ -0,0 +1,65 @@
+function _wait-instance-connectable() {
+  for attempt in $(seq 12); do
+    if ! ssh -qo StrictHostKeyChecking=no "ubuntu@$ip_addr" -- true; then
```
Comment (toabctl) [inline scripts/ubuntu-bartender/azure-provider:3]: how can I set the private key when using `AZURE_PUBKEY_PATH` ? the aws-provider adds some `-i ...` and just assumes that there might be a local file . Maybe I'm missing something here?

**Review by toabctl — COMMENTED** [review]
Just one question - beside that, looks good!

File: scripts/ubuntu-bartender/azure-provider (hunk 1 above)
Comment (gjolly) [reply scripts/ubuntu-bartender/azure-provider:3]: Yes, I remember now and I overlooked this part.

Essentially, it is not possible on Azure to do the same as on AWS. Azure (from what I know) doesn't let you store your keys and re-use them on VMs[1].

[1] https://docs.microsoft.com/en-us/azure/virtual-machines/linux/create-ssh-keys-detailed

Comment (rcj4747) [reply scripts/ubuntu-bartender/azure-provider:3]: You can now [store SSH keys in Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/ssh-keys-portal) in the portal.  It's not fully integrated into their instructions as your link doesn't show this code path, but it works.

Comment (gjolly) [reply scripts/ubuntu-bartender/azure-provider]: Nice to see you here @rcj4747 , I pushed an alternative solution that I prefer. But I can also revert it an use this feature if needed.

**Review by toabctl — APPROVED** [review]

**Pull request merged**